core/db/db.go

```diff
@@ -16,18 +19,19 @@ type InstanceID string
 
 // NewInstanceID generates a new identity for an instance.
 func NewInstanceID() InstanceID {
-	return InstanceID(uuid.New().String())
+	t := time.Now().UTC()
+	entropy := ulid.Monotonic(rand.New(rand.NewSource(t.UnixNano())), 0)
+	id := ulid.MustNew(ulid.Timestamp(t), entropy)
+	return InstanceID(id.String())
```

Comment:
The api for this go ulid lib is kinda ugly. I'm not quite following what the docs [here](https://github.com/oklog/ulid#usage) mean about the thread safety issue. The way I've implemented this method, it's creating a new `rand.Rand` each time this method is called. Input welcome here.

Reply:
I think he's referring to the `math/rand` package specifically as opposed to `crypto/rand`. Looking at the tests, you could do something simpler like:

```
id := ulid.MustNew(ulid.Now(), rand.Reader)
```

Where `rand` is from `crypto/rand`. Guessing `ulid.Now()` returns now in millis.